pantry-0.10.0 causing breakages to mega-sdist

```
       [2 of 2] Compiling Main [Pantry package changed]                                                                                                                                                             
                                                                                                                                                                                                                    
       /home/curators/work/unpack-dir/unpacked/mega-sdist-0.4.3.0-ec104f72402fffb856cce7e19134b524c1f0a7cf6484f12c5d38f0c8af7c5cc4/mega-sdist.hs:97:7: error: [GHC-83865]
           • Couldn't match expected type: RIO LogFunc ()
                         with actual type: (PantryConfig -> RIO env0 a0) -> RIO env0 a0                                                                                                                             
           • Probable cause: ‘($)’ is applied to too few arguments
             In the second argument of ‘($)’, namely
               ‘withPantryConfig                                                                          
                  root defaultPackageIndexConfig HpackBundled 8 defaultCasaRepoPrefix
                  defaultCasaMaxPerRequest defaultSnapshotLocation
                  $ \ pantryConfig       
                      -> do let ...                 
                            runRIO app $ main2 args’
             In the expression:
               runRIO lf                        
                 $ withPantryConfig
                     root defaultPackageIndexConfig HpackBundled 8 defaultCasaRepoPrefix
                     defaultCasaMaxPerRequest defaultSnapshotLocation
                     $ \ pantryConfig       
                         -> do let ...                                                                                                                                                                              
                               runRIO app $ main2 args
             In the second argument of ‘($)’, namely
               ‘\ rm           
                  -> runRIO lf
                       $ withPantryConfig
                           root defaultPackageIndexConfig HpackBundled 8 defaultCasaRepoPrefix
                           defaultCasaMaxPerRequest defaultSnapshotLocation
                           $ \ pantryConfig -> do ...’
          |
       97 |       withPantryConfig
          |       ^^^^^^^^^^^^^^^^...

       /home/curators/work/unpack-dir/unpacked/mega-sdist-0.4.3.0-ec104f72402fffb856cce7e19134b524c1f0a7cf6484f12c5d38f0c8af7c5cc4/mega-sdist.hs:112:39: error: [GHC-83865]
           • Couldn't match expected type ‘PantryConfig’
                         with actual type ‘WantedCompiler’
           • In the ‘appPantryConfig’ field of a record
             In the expression:
               App
                 {appLogFunc = lf, appProcessContext = pc, appResourceMap = rm,
                  appPantryConfig = pantryConfig}
             In an equation for ‘app’:
                 app
                   = App
                       {appLogFunc = lf, appProcessContext = pc, appResourceMap = rm,
                        appPantryConfig = pantryConfig}
           |
       112 |                   , appPantryConfig = pantryConfig
           |                                       ^^^^^^^^^^^^
       
       /home/curators/work/unpack-dir/unpacked/mega-sdist-0.4.3.0-ec104f72402fffb856cce7e19134b524c1f0a7cf6484f12c5d38f0c8af7c5cc4/mega-sdist.hs:114:13: error: [GHC-83865]
           • Couldn't match expected type ‘GlobalHintsLocation’
                         with actual type ‘m0 ()’
           • In a stmt of a 'do' block: runRIO app $ main2 args
             In the expression:
               do let app = ...
                  runRIO app $ main2 args
             In the second argument of ‘($)’, namely
               ‘\ pantryConfig
                  -> do let ...
                        runRIO app $ main2 args’
           |
       114 |             runRIO app $ main2 args
           |             ^^^^^^^^^^^^^^^^^^^^^^^
```

Will add an upper bound to `pantry` for now. The only breakage identified so far is in `mega-sdist`.

CC @snoyberg

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -922,7 +922,6 @@ packages:
 
         - time-manager
         - pantry < 0.10.0 # https://github.com/commercialhaskell/stackage/issues/7360
-        - mega-sdist
         - http-download
         - hi-file-parser
         - rio-prettyprint

